BP/common/casus_belli_types/00_dejure_war.txt

```diff
@@ -0,0 +1,1758 @@
+		}
+		setup_de_jure_cb = {
+			attacker = scope:attacker
+			defender = scope:defender
+			victory = no
+			title = scope:target
+		}
+		# Prestige loss for the attacker
+		scope:attacker = {
+			add_prestige = {
+				value = scope:cb_prestige_factor # Set by 'setup_de_jure_cb'
+				multiply = -5.0
+			}
+		}
+		# Allies on both sides get full prestige value for helping in the war, based on their war participation.
+		modify_allies_of_participants_fame_values = {
+			WINNER = scope:attacker
+			FAME_BASE = scope:cb_prestige_factor # Set by 'setup_de_jure_cb'
+			IS_RELIGIOUS_WAR = no
+			WINNER_ALLY_FAME_SCALE = 100
+			LOSER_ALLY_FAME_SCALE = 100
+		}
+		# Defender neither gains nor loses any prestige.
+		
+		# Truce
+		add_truce_white_peace_effect = yes
+	}
+
+	on_defeat_desc = {
+		first_valid = {
+			triggered_desc = {
+				trigger = { scope:defender = { is_local_player = yes } }
+				desc = de_jure_cb_defeat_desc_defender
+			}
+			triggered_desc = {
+				trigger = { scope:attacker = { is_local_player = yes } }
+				desc = de_jure_cb_defeat_desc_attacker
+			}
+			desc = de_jure_cb_defeat_desc
+		}
+
+		
+	}
+	
+	on_defeat = {
+		scope:attacker = { show_pow_release_message_effect = yes }
+		# Attacker must pay the defender war reparations.
+		scope:attacker = {
+			pay_short_term_gold = {
+				gold = 3
+				target = scope:defender
+				yearly_income = yes
+			}
+		}
+
+		# Create an 'helper' de jure CB which we won't execute, but can use to determine how much Prestige should be given to war participants.
+		every_in_list = {
+			list = target_titles
+			save_temporary_scope_as = target
+		}
+		setup_de_jure_cb = {
+			attacker = scope:attacker
+			defender = scope:defender
+			victory = no
+			title = scope:target
+		}
+		# Attacker loses Prestige, all other participants gain Prestige based on their war contribution (Defender gets full Prestige).
+		modify_all_participants_fame_values = {
+			WINNER = scope:defender
+			LOSER = scope:attacker
+			FAME_BASE = scope:cb_prestige_factor # Set by 'setup_de_jure_cb'
+			IS_RELIGIOUS_WAR = no
+			WINNER_FAME_SCALE = 10
+			LOSER_FAME_SCALE = -10
+			WINNER_ALLY_FAME_SCALE = 100
+			LOSER_ALLY_FAME_SCALE = 100
+		}
+	
+		# Truce
+		add_truce_attacker_defeat_effect = yes
+		
+		scope:attacker = {
+			save_temporary_scope_as = loser
+		}
+		on_lost_aggression_war_discontent_loss = yes
+	}
+	
+	on_primary_attacker_death = inherit
+	on_primary_defender_death = inherit
+	
+	transfer_behavior = transfer
+	
+	attacker_allies_inherit = yes
+	defender_allies_inherit = yes
+	
+	war_name = "DEJURE_CB_WAR_SPECIFIC_NAME"
+	war_name_base = "DEJURE_CB_WAR_SPECIFIC_NAME_BASE"
+	cb_name = DEJURE_CB_DUCHY_NAME
+	cb_name_no_target = DEJURE_CB_DUCHY_NAME_GENERIC
+	interface_priority = 99 # Below the "take all" one
+	
+	attacker_wargoal_percentage = 0.8
+	
+	max_ai_diplo_distance_to_title = 500
+	min_ai_score = 100
+}
+
+imperial_reconquest_cb = { #TODO: Rework in context of E/W Roman Empire and Roman Successors
```

Comment:
We already reworked this CB in CK2.

Reply:
Correct; I haven't ported it in yet because I haven't defined all the new regions on the map that go with it. Reason:  that seemed to require refactoring de jure titles first before fully implementing.

Reply:
Yes, no problems, I just wanted to point out there was a skeleton. 😉